tests/fixtures/expanders.php

```diff
@@ -13,3 +13,63 @@ function my_hello_expander(TokenStream $ts) : TokenStream {
         )
     ;
 }
+
+function reverse_ast_expander(ParsedTokenStream $stream, Engine $engine) : ParsedTokenStream {
+    $ast = $stream->getAst();
+
+    $tokens = $ast->tokens();
+    $first = array_shift($tokens);
+
+    _modifyPrivateProperty($ast, "ast", function ($value) use ($first) {
+        $value["outer"]["inner"][0] = new Token($first->type(), strrev($first->value()));
+        return $value;
+    });
```

Comment:
```diff
-$tokens = $ast->tokens();
-$first = array_shift($tokens);
-
-_modifyPrivateProperty($ast, "ast", function ($value) use ($first) {
-    $value["outer"]["inner"][0] = new Token($first->type(), strtoupper($first->value()));
-    return $value;
-});
+$leaf = $ast->{'* outer inner 0'}->token();
+new Ast($ast->label(), ['outer' => ['inner' => new Token($leaf->type(), strrev($leaf))]]);
```

Reply:
I think this is splitting hairs at this point. Why does it matter how the AST is modified?

Reply:
This doesn't work, see comments above.